src/Storages/StorageFile.cpp

```diff
@@ -626,8 +674,29 @@ class StorageFileSource : public ISource
                 }
 
                 if (!read_buf)
-                    read_buf = createReadBuffer(current_path, storage->use_table_fd, storage->getName(), storage->table_fd, storage->compression_method, context);
-
+                {
+                    if (files_info->paths_to_archive.empty())
```

Comment:
```suggestion
                    if (files_info->paths_to_archive.empty())
```

I would create function with this check to have it more clear, something like `fileInArchive`